### PR TITLE
feat(cgo): Implement graceful shutdown

### PR DESCRIPTION
**Pull request by Fangop**
* The code has been formatted by `gofmt`
* The update of `go.sum` and `go.mod` has been synchronized by `go mod tidy`
* The graceful shutdown of dcurl cgo module will destroy dcurl while receiving `SIGTERM` and `SIGINT` signal as [HORNET](https://github.com/gohornet/hornet/blob/v0.4.1/plugins/gracefulshutdown/plugin.go).
* When destroying dcurl, the logger will log the warning of destroying dcurl.

## Graceful Shutdown
The graceful shutdown implementation requires 2 new go package to be imported.
* The package `syscall` is for the information of system call.
* The package `os` and its sub package `os/signal` is for notifying the signal of system
The concept of the implementation is to start a goroutine when importing dcurl module.
The goroutine will listen to signal of the system.
When receiving `SIGTERM` and `SIGINT`, it'll log the warning and safely destroy dcurl.

## Logger 
The logger is based on the primitive log package in go.
Originally, I use the `go.uber.org/zap`(the logger `iotaledger/hive.go/logger` based on) for the logger.
Consider of the complexity of the synchronization of the loggers, I used primitive log package in go for its simplicity.

Thanks for the reviewing!

### CODE REVIEW

File: cgo/pow/pow.go [hunk 1]
```diff
@@ -36,6 +40,15 @@ func init() {
 	C.dcurl_init_wrapper()
 	proofOfWorkFuncs["Dcurl"] = DcurlProofOfWork
 	proofOfWorkFuncs["SyncDcurl"] = SyncDcurlProofOfWork
+
+	logger := log.New(os.Stdout, "WARN", log.Ldate | log.Ltime)
+	gracefulShutdown := make(chan os.Signal)
+	signal.Notify(gracefulShutdown, syscall.SIGTERM, syscall.SIGINT)
+	go func() {
+		<- gracefulShutdown
+		logger.Println("test")
```
Comment (junwei0117) [inline cgo/pow/pow.go]: Remove testing message.

Comment (Fangop) [reply cgo/pow/pow.go]: It has been modified to warning message.
Thanks for your  review!

File: cgo/pow/pow.go [hunk 2]
```diff
@@ -18,8 +18,12 @@ char *dcurl_wrapper(char* trytes, int mwm, int threads){
 */
 import "C"
 import (
+	"os"
+	"os/signal"
+	"log"
 	"sync"
 	"unsafe"
+	"syscall"
```
Comment (marktwtn) [inline cgo/pow/pow.go]: Change to alphabetical order.

**Review by marktwtn — CHANGES_REQUESTED** [review]

File: cgo/pow/pow.go (hunk 2 above)
Comment (Fangop) [reply cgo/pow/pow.go]: OK! The request has been done!

**Review by marktwtn — APPROVED** [review]

**Pull request merged**